portal-api/src/v1/users/controller.js

```diff
@@ -256,20 +270,23 @@ exports.updateLastLogin = async (user, sessionIdentifier, callback = () => {}) =
   callback();
 };
 
+// TODO DTFS2-6680: move this function to use new user service/repo
 exports.incrementFailedLoginCount = async (user) => {
   if (!ObjectId.isValid(user._id)) {
-    throw new Error('Invalid User Id');
+    throw new InvalidUserIdError(user._id);
   }
 
   const failureCount = user.loginFailureCount ? user.loginFailureCount + 1 : 1;
   const thresholdReached = failureCount >= businessRules.loginFailureCount_Limit;
 
   const collection = await db.getCollection('users');
-  const update = {
-    loginFailureCount: failureCount,
-    lastLoginFailure: now(),
-    'user-status': thresholdReached ? USER.STATUS.BLOCKED : user['user-status'],
-  };
+  let update = { loginFailureCount: failureCount, lastLoginFailure: now() };
+  if (thresholdReached) {
+    update = {
+      'user-status': USER.STATUS.BLOCKED,
+      'blocked-status-reason': USER.STATUS_BLOCKED_REASON.INVALID_PASSWORD,
+    };
+  }
```

Comment:
```suggestion
  const update = thresholdReached
    ? {
      'user-status': USER.STATUS.BLOCKED,
      'blocked-status-reason': USER.STATUS_BLOCKED_REASON.INVALID_PASSWORD,
    } : {
      loginFailureCount: failureCount,
      lastLoginFailure: now()
    };
```

I think this is less likely to cause bugs in future as it's clearer that `update` is one or the other